core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -483,7 +483,24 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
               provide(fullyQualifiedActionName(actionName)) { fullActionName =>
                 onComplete(verifyWebAction(fullActionName, onBehalfOf.isDefined)) {
                   case Success((actionOwnerIdentity, action)) =>
-                    if (!action.annotations.getAs[Boolean]("web-custom-options").exists(identity)) {
+                    val requireWebAuthIsBool = (action.annotations.getAs[Boolean]("require-whisk-auth") != None)
+                    val annotationRequireWebAuthIsIntOrString = ((action.annotations
+                      .getAs[String]("require-whisk-auth") != None) || (action.annotations
+                      .getAs[Int]("require-whisk-auth") != None))
+                    val annotationRequireWebAuth = (action.annotations
+                      .getAs[Int]("require-whisk-auth")
+                      .getOrElse(action.annotations.getAs[String]("require-whisk-auth").getOrElse("")))
+                      .toString
+                    val enforceWhiskAuthHdr = (!requireWebAuthIsBool && annotationRequireWebAuthIsIntOrString)
+                    val headerWhiskAuthSeq = context.headers.filter(_.lowercaseName == "x-require-whisk-auth")
+                    // If the require-whisk-auth annotation is either an integer or a string, secure the web action by enforcing that
+                    // the require-whisk-auth annotation value == request header x-require-whisk-auth value
+                    if (enforceWhiskAuthHdr && (headerWhiskAuthSeq.isEmpty || headerWhiskAuthSeq(0).value != annotationRequireWebAuth)) {
```

Comment:
With the `find => Option[Header]` change thise becomes obsolete, but for posterity: Sequences, Lists etc. generally have a `.headOption` method, which returns `Some[A]` if the sequence is not empty or `None` if it is.